### PR TITLE
Add SP pagination and search feature

### PR DESCRIPTION
**Pull request by LahiruLS**
Resolves: https://github.com/wso2/product-is/issues/5280

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/org/wso2/carbon/identity/application/mgt/ui/i18n/Resources.properties [hunk 1]
```diff
@@ -273,5 +273,7 @@ alert.error.default.seq.name.invalid=Provided default authentication sequence na
 enter.service.provider.name.pattern=Enter service provider name pattern (* for all)
 service.provider.search=Search
 config.application.JWKS = JWKS URI
+enter.service.provider.name.pattern=Enter service provider name pattern (* for all)
```
Comment (madurangasiriwardena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/org/wso2/carbon/identity/application/mgt/ui/i18n/Resources.properties]: Aren't these properties already defined?

Comment (LahiruLS) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/org/wso2/carbon/identity/application/mgt/ui/i18n/Resources.properties]: remove the unnecessary line

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtDBQueries.java [hunk 2]
```diff
@@ -80,9 +80,36 @@
     // LOAD Queries
     public static final String LOAD_APP_ID_BY_APP_NAME = "SELECT ID FROM SP_APP WHERE APP_NAME = ? AND TENANT_ID = ?";
     public static final String LOAD_APP_NAMES_BY_TENANT = "SELECT ID, APP_NAME, DESCRIPTION FROM SP_APP WHERE " +
-            "TENANT_ID = ?";
-    public static final String LOAD_APP_NAMES_BY_TENANT_AND_APP_NAME = "SELECT ID, APP_NAME, DESCRIPTION FROM SP_APP " +
-            "WHERE TENANT_ID = ? AND APP_NAME LIKE ?";
+            "TENANT_ID = ? ORDER BY ID DESC";
+    public static final String LOAD_APP_NAMES_BY_TENANT_AND_APP_NAME = "SELECT APP_NAME, DESCRIPTION FROM SP_APP WHERE " +
```
Comment (GDRDABARERA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtDBQueries.java]: Don't  we have querries per each db type?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 3]
```diff
@@ -239,13 +240,141 @@ public ServiceProvider getApplicationExcludingFileBasedSPs(String applicationNam
...
+     * Get All Application Basic Information with pagination
+     *
+     * @param tenantDomain Tenant Domain
+     * @param username     User Name
+     * @param pageNumber   Number of the page
+     * @return ApplicationBasicInfo[]
+     * @throws IdentityApplicationManagementException
+     */
+    @Override
+    public ApplicationBasicInfo[] getAllPaginatedApplicationBasicInfo(String tenantDomain, String username, int pageNumber) throws IdentityApplicationManagementException {
```
Comment (GDRDABARERA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: shall we add some debug logs..

**Pull request merged**